BridgeAppSDK/DataArchiving/SBADataArchive.m

```diff
@@ -0,0 +1,280 @@
+}
+
+- (void)insertURLIntoArchive:(NSURL*)url fileName:(NSString *)filename
+{
+    NSData *dataToInsert = [NSData dataWithContentsOfURL:url];
+    [self insertDataIntoArchive:dataToInsert filename:filename];
+}
+
+- (void)insertDictionaryIntoArchive:(NSDictionary *)dictionary filename:(NSString *)filename
+{
+    NSError * serializationError;
+    NSData * jsonData = [NSJSONSerialization dataWithJSONObject:dictionary options:NSJSONWritingPrettyPrinted error:&serializationError];
+        
+    if (jsonData != nil) {
+        [self insertDataIntoArchive:jsonData filename:filename];
+    }
+    else {
+        SBALogError2(serializationError);
+        NSAssert(NO, @"Failed to serialize JSON dictionary");
+    }
+}
+
+- (void)insertDataIntoArchive :(NSData *)data filename: (NSString *)filename
+{
+    [self.zipEntries addObject: [ZZArchiveEntry archiveEntryWithFileName: filename
+                                                                compress:YES
+                                                               dataBlock:^(NSError** error)
+                                 {
+                                     SBALogError2(*error);
+                                     return data;
+                                 }]];
+    
+    //add the fileInfoEntry
+    NSString *extension = [filename pathExtension] ? : kJsonPathExtension;
+    NSDictionary *fileInfoEntry = @{ kFileInfoNameKey: filename,
+                                     kFileInfoTimeStampKey: [[NSDate date] jsonObjectWithFormatter:nil],
+                                     kFileInfoContentTypeKey: [self contentTypeForFileExtension:extension] };
+    
+    [self.filesList addObject:fileInfoEntry];
+
+}
+
++ (NSString *)appVersion
+{
+    static NSString *appVersion = nil;
+    if (!appVersion)
+    {
+        NSString *version = [NSBundle.mainBundle objectForInfoDictionaryKey:@"CFBundleShortVersionString"];
+        NSString *build   = NSBundle.mainBundle.appVersion;
+        appVersion	= [NSString stringWithFormat: @"version %@, build %@", version, build];
+    }
+    
+    return appVersion;
+}
+
+
+//Compiles the final info.json file and inserts it into the zip archive.
+-(NSError *)completeArchive
+{
+    NSError *error = nil;
+    if (self.filesList.count) {
+        [self.infoDict setObject:self.filesList forKey:kFilesKey];
+        
+        [self.infoDict setObject:self.reference forKey:kItemKey];
+        [self.infoDict setObject:[[NSBundle mainBundle] appName] forKey:kAppNameKey];
+        [self.infoDict setObject:[self.class appVersion] forKey:kAppVersionKey];
+        [self.infoDict setObject:[[UIDevice currentDevice] deviceInfo] forKey:kPhoneInfoKey];
+
+        // TODO: syoung 04/26/2016 Implement using setArchiveInfoObject: to add these values
+
+//        [self.infoDict setObject:[NSUUID new].UUIDString forKey:kTaskRunKey];
+        if (self.schemaRevision) {
+            [self.infoDict setObject:self.schemaRevision forKey:kSchemaRevisionKey];
+        }
```

Comment:
Use the setArchiveInfoObject:forKey: method instead which can be used to add any custom info to the dictionary.